experiment/run_experiment.py

```diff
@@ -382,14 +389,14 @@ def start(self):
             '-e',
             set_experiment_arg,
             '-e',
-            set_cloud_project_arg,
-            '-e',
             sql_database_arg,
             '-e',
             set_experiment_filestore_arg,
             '-e',
             set_report_filestore_arg,
             '-e',
+            set_docker_registry_arg,
+            '-e',
```

Comment:
I don't think we need to set this env var. Really the only config vars that should be set in the env are ones that are used in startup-dispatcher.sh.

Reply:
Need to test. Probably we should keep it. Since cloud_project is passed to runner docker containers level, we should make sure the only usage of them is not related to docker_registry construction logic in cloud settings.

Reply:
Will defer to Jonathan on this part of review.

Reply:
Hi, I tried one experiment running without setting it but failed to build images.
Probably it is because `experiment/dispatcher.py` imports `experiment/build/builder.py`, which imports `experiment/build/local_build.py`, where `docker_registry` env is used.

Maybe we keep it as this way now?

Reply:
What's the error?
It's clear https://github.com/google/fuzzbench/pull/461/files#diff-b6cb9bdd02ffa26364e101706ffa0fc1R71 will break unless it is modified. The questions is whether we should modify it to support config file or not.
I only set configs as env vars if they are needed in the startup script otherwise I read them from the config file and pass them through to any functions that need them. This breaks that policy.
It may be fine to change this policy (we could make the code that reads the config set env vars for each config var) but if we do that it should be an intentional decision and we should do it all at once, not just in one place.

Reply:
> @lszekeres and @inferno-chromium I'd like your opinion on this: Should we continue the policy that (almost all) config variables are read from the config file and passed to functions that use them? Or could we change things to so that we set env vars for each config var. I think the latter would make code easier to write and shorter to read but possibly harder to understand.

Yes, in general, whenever possible, global state such as environment vars should be avoided, as it makes the code harder to reason about and maintain.

(Re. this particular line of code, i don't have enough context, so don't know what is the best thing to do here).

Reply:
> > @lszekeres and @inferno-chromium I'd like your opinion on this: Should we continue the policy that (almost all) config variables are read from the config file and passed to functions that use them? Or could we change things to so that we set env vars for each config var. I think the latter would make code easier to write and shorter to read but possibly harder to understand.
> 
> Yes, in general, whenever possible, global state such as environment vars should be avoided, as it makes the code harder to reason about and maintain.
> 
> (Re. this particular line of code, i don't have enough context, so don't know what is the best thing to do here).

TODO added and issues created for this #484.
Thanks for this discussion.